tgui/packages/tgui/interfaces/CommunicationsConsole.js

```diff
@@ -0,0 +1,712 @@
+      tooltipPosition="bottom-right"
+      onClick={() => act("recallShuttle")}
+    />);
+  } else {
+    emergencyShuttleButton = (<Button
+      icon="space-shuttle"
+      content="Call Emergency Shuttle"
+      disabled={shuttleCanEvacOrFailReason !== 1}
+      tooltip={
+        shuttleCanEvacOrFailReason !== 1
+          ? shuttleCanEvacOrFailReason
+          : undefined
+      }
+      tooltipPosition="bottom-right"
+      onClick={() => setCallingShuttle(true)}
+    />);
+  }
+
+  if (canBuyShuttles !== 0) {
+    generalFunctions.push(<Button
+      icon="shopping-cart"
+      content="Purchase Shuttle"
+      disabled={canBuyShuttles !== 1}
+      // canBuyShuttles is a string detailing the fail reason
+      // if one can be given
+      tooltip={canBuyShuttles !== 1 ? canBuyShuttles : undefined}
+      tooltipPosition="right"
+      onClick={() => act("setState", { state: STATE_BUYING_SHUTTLE })}
+    />);
+  }
+
+  if (canMessageAssociates) {
+    generalFunctions.push(<Button
+      icon="comment-o"
+      content={`Send message to ${emagged ? "[UNKNOWN]" : "CentCom"}`}
+      disabled={!importantActionReady}
+      onClick={() => setMessagingAssociates(true)}
+    />);
+
+    if (messagingAssociates) {
+      children.push(<MessageModal
+        label={`Message to transmit to ${emagged ? "[ABNORMAL ROUTING COORDINATES]" : "CentCom"} via quantum entanglement`}
+        notice="Please be aware that this process is very expensive, and abuse will lead to...termination. Transmission does not guarantee a response."
+        icon="bullhorn"
+        buttonText="Send"
+        onBack={() => setMessagingAssociates(false)}
+        onSubmit={message => {
+          setMessagingAssociates(false);
+          act("messageAssociates", {
+            message,
+          });
+        }}
+      />);
+    }
+  }
+
+  if (canRequestNuke) {
+    generalFunctions.push(<Button
+      icon="bomb"
+      content="Request Nuclear Authentication Codes"
+      disabled={!importantActionReady}
+      onClick={() => setRequestingNukeCodes(true)}
+    />);
+
+    if (requestingNukeCodes) {
+      children.push(<MessageModal
+        label="Reason for requesting nuclear self-destruct codes"
+        notice="Misuse of the nuclear request system will not be tolerated under any circumstances. Transmission does not guarantee a response."
+        icon="bomb"
+        buttonText="Request Codes"
+        onBack={() => setRequestingNukeCodes(false)}
+        onSubmit={reason => {
+          setRequestingNukeCodes(false);
+          act("requestNukeCodes", {
+            reason,
+          });
+        }}
+      />);
+    }
+  }
+
+  if (emagged) {
+    generalFunctions.push(<Button
+      icon="undo"
+      content="Restore Backup Routing Data"
+      onClick={() => act("restoreBackupRoutingData")}
+    />);
+  }
+
+  if (callingShuttle) {
+    children.push(<MessageModal
```

Comment:
Ok let me tell you, pushing children into arrays like this is ugly for two reasons:

1. It's more code, because you have to initialize the list, then waste two more lines on the `.push` call.
2. Each element would require a unique key, because since it's an array, inferno has no way of knowing which array element is which, which can create all sorts of bugs.

Normal way of inlining logic has no such problems, i.e.:

```jsx
<div>
  {condition1 && (
    <Foo />
  )}
  {condition2 && (
    <Bar />
  )}
</div>
```

Why? Because on false condition, that particular child is not removed from the list, but is instead a falsy value, which is not rendered. This does not require keys because order is static and never changes.